Sources/AutomergeRepo/Networking/Providers/WebSocketProvider.swift

```diff
@@ -394,10 +394,50 @@ public final class WebSocketProvider: NetworkProvider {
                         )
                 }
                 reconnectAttempts += 1
+                                
                 do {
-                    try await Task.sleep(for: .seconds(waitBeforeReconnect))
-                    // if endpoint is nil, this returns nil
-                    _ = try await attemptConnect(to: endpoint)
+                    // Wait to reconnect. Wait for waitBeforeReconnect and networth path
+                    // transitioning from not satisfied to satisfied. Whichever comes first.
+                    let success = try await withThrowingTaskGroup(of: Void.self) { group in
+
+                        // Wait for timeout
+                        group.addTask {
+                            try await Task.sleep(for: .seconds(waitBeforeReconnect))
+                        }
+                        
+                        // Wait for network becomming availible
+                        group.addTask {
+                            let monitor = NWPathMonitor()
```

Comment:
I don't know how "heavy" it is to create NWPathMonitor, but that may be something we don't want to do in a loop that iterates through with every reconnect need. Maybe that's not an issue, but it's something I'd tend to try and initialize outside of any looping structure and use as needed inside the loop. Is it possible to initialize it externally in the class, get the current value when we want to start checking, and then set up the AsyncStream to provide updates as they're invoked from callbacks? My goal with that question is just being worried, without any real knowledge, about what it means to create that instance and uninitialize it (as it falls out of scope), in the reconnect loop flow.